IntoOOP/Complex.cs

```diff
@@ -0,0 +1,245 @@
+
+    /// <summary>Оператор инкремента комплексных чисел.</summary>
+    /// <param name="value">Операнд.</param>
+    /// <returns>Прибавляет 1 к действительной части операнда.</returns>
+    public static Complex operator ++(Complex value) => value + 1;
+
+    /// <summary>Унарный оператор вычитания.</summary>
+    /// <param name="value">Операнд.</param>
+    /// <returns>Умножает значение опернда на -1.</returns>
+    public static Complex operator -(Complex value) => value * -1;
+
+    /// <summary>Бинарный оператор вычитания.</summary>
+    /// <param name="left">Левый операнд.</param>
+    /// <param name="right">Правый операнд.</param>
+    /// <returns>Разность операндов.</returns>
+    public static Complex operator -(Complex left, Complex right) => new Complex(left.R - right.R, left.I - right.I);
+
+    /// <summary>Оператор декремента комплексных чисел.</summary>
+    /// <param name="value">Операнд.</param>
+    /// <returns>Вычитает 1 из действительной части операнда.</returns>
+    public static Complex operator --(Complex value) => value - 1;
+
+    /// <summary>Оператор умножения комплексных чисел.</summary>
+    /// <param name="left">Левый операнд.</param>
+    /// <param name="right">Правый операнд.</param>
+    /// <returns>Результат умножения операндов.</returns>
+    public static Complex operator *(Complex left, Complex right)
+    {
+        var r = left.R * right.R;
+        var i = left.I * right.I;
+        return new Complex(r - i, r + i);
+    }
+
+    /// <summary>Оператор деления комплексных чисел.</summary>
+    /// <param name="left">Левый операнд.</param>
+    /// <param name="right">Правый операнд.</param>
+    /// <returns>Результат деления операндов.</returns>
+    /// <exception cref="DivideByZeroException">Возбуждается, если действительная и мнимая части 
+    /// правого операнда равны 0..</exception>
+    public static Complex operator /(Complex left, Complex right)
+    {
+        if (right.R == 0 && right.I == 0)
+            throw new DivideByZeroException();
+
+        var d = right.R * right.R + right.I * right.I;
+
+        return new Complex((left.R * right.R + left.I * right.I) / d, (right.R * left.I - left.R * right.I) / d);
+    }
+
+    /// <summary>Оператор равенства комплексных чисел.</summary>
+    /// <param name="r1">Левый операнд.</param>
+    /// <param name="r2">Правый операнд.</param>
+    /// <returns>Результат проверки на равенство комплексных чисел.</returns>
+    public static bool operator ==(Complex left, Complex right) => left.Equals(right);
+
+
+    /// <summary>Оператор неравенства комплексных чисел.</summary>
+    /// <param name="r1">Левый операнд.</param>
+    /// <param name="r2">Правый операнд.</param>
+    /// <returns>Результат проверки на неравенство комплексных чисел.</returns>
+    public static bool operator !=(Complex left, Complex right) => !left.Equals(right);
```

Comment:
```suggestion
    public static bool operator !=(Complex left, Complex right) => !(left == right);
```
Иначе если left будет null, то всё упадёт.